Add Against All Odds server midnight wait timer

Era-Lusiphur opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to Topaz Next's [Limited Contributor License Agreement](https://github.com/DerpyProjectGroup/topaz/blob/info/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/DerpyProjectGroup/topaz/blob/info/CONTRIBUTING.md)**
- [ ] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Had an issue with this NPC elsewhere, discovered the timer wasn't in place for the JP midnight wait for re-acquiring another KI for the battle. This adds a timer on the initial CS, adds a check for it but also accounts for characters that didn't get it set previously on the NPC to re-acquire the KI, and clears it on the battlefield victory.

https://www.bg-wiki.com/ffxi/Against_All_Odds
https://ffxiclopedia.fandom.com/wiki/Against_All_Odds

Era-Lusiphur committed:
Add Against All Odds server midnight wait timer

Had an issue with this NPC elsewhere, discovered the timer wasn't in place for the JP midnight wait for re-acquiring another KI for the battle. This adds a timer on the initial CS, adds a check for it but also accounts for characters that didn't get it set previously on the NPC to re-acquire the KI, and clears it on the battlefield victory.

https://www.bg-wiki.com/ffxi/Against_All_Odds
https://ffxiclopedia.fandom.com/wiki/Against_All_Odds

Era-Lusiphur committed:
Correct var names

I had renamed the var for PR so that it was more obvious but forgot to change it in every spot. Whoops.

Era-Lusiphur committed:
Update Ratihb.lua

claywar commented on scripts/zones/Aht_Urhgan_Whitegate/npcs/Ratihb.lua:
```diff
@@ -24,8 +24,8 @@ entity.onTrigger = function(player, npc)
         player:startEvent(772)
-    elseif againstAllOdds == QUEST_ACCEPTED and not player:hasKeyItem(xi.ki.LIFE_FLOAT) then
-        player:startEvent(604)
+    if (player:getCharVar("AgainstAllOdds") == 2 and (player:getCharVar("AgainstAllOddsTimer") < os.time() or player:getCharVar("AgainstAllOddsTimer") == 0)) then
```
Could you please remove the outer parentheses for styling?

Era-Lusiphur replied:
Beaten to it!

Xaver-DaRed committed:
style change

Xaver-DaRed reviewed (approved):
Took the liberty to update it for you, lusi.

Era-Lusiphur committed:
Update Ratihb.lua

Forgot a piece from porting over. Should be setting timer back again to next JP midnight on awarding KI, should player fail a 2nd time.
